frontend/src/api/k8s/secrets.ts

```diff
@@ -157,3 +159,78 @@ export const deleteSecret = (
       queryOptions: { name: secretName, ns: projectName },
     }),
   );
+
+export const assembleEdgeS3Secret = (
+  projectName: string,
+  data: Map<string, string>,
+  secretName?: string,
+): SecretKind => {
+  const name = `credentials-s3`;
+
+  const storageConfigData = {
+    type: 's3',
+    access_key_id: data.get(AWS_KEYS.ACCESS_KEY_ID),
+    secret_access_key: data.get(AWS_KEYS.SECRET_ACCESS_KEY),
+    endpoint_url: data.get(AWS_KEYS.S3_ENDPOINT),
+    region: data.get(AWS_KEYS.DEFAULT_REGION),
+  };
+
+  return {
+    apiVersion: 'v1',
+    kind: 'Secret',
+    metadata: {
+      name: secretName || name,
+      namespace: projectName,
+    },
+    stringData: {
+      'aws-storage-config': JSON.stringify(storageConfigData),
+    },
+  };
+};
+
+export const createEdgeS3Secrets = (data: SecretKind, opts?: K8sAPIOptions): Promise<SecretKind> =>
+  k8sCreateResource<SecretKind>(
+    applyK8sAPIOptions(opts, {
+      model: EdgeSecretModel,
+      resource: data,
+    }),
+  );
+
+export const assembleEdgeImageRegistrySecret = (
+  projectName: string,
+  data: EdgeModelState,
+  secretName?: string,
+): SecretKind => {
+  const name = `credentials-quay-image-registry`;
```

Comment:
same as above, we should not hard code this. we need to make the name dynamic to the model